lib/services/elasticsearch.js

```diff
@@ -687,6 +687,65 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+  * Updates all documents matching the provided filters
+  *
+  * @param {String} index - Index name
+  * @param {String} collection - Collection name
+  * @param {Object} query - Query to match documents
+  * @param {Object} options - from (undefined), size (undefined), refresh (undefined)
+  *
+  * @returns {Promise.<Object>} { documents, total, updated, failures: [ _shardId, reason ] }
+  */
+  async updateByQuery(
+    index,
+    collection,
+    query,
+    changes,
+    { refresh, from, size = 1000 } = {})
+  {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: this._sanitizeSearchBody({ query }),
+      scroll: '5s',
+      from,
+      size
+    };
+
+    if (!isPlainObject(query)) {
+      throw errorsManager.get('missing_argument', 'body.query');
+    }
+
+    try {
+      const documents = await this._getAllDocumentsFromQuery(esRequest);
+      documents.forEach(document => {
+        delete document._source;
+        document.body = changes;
+        return document;
+      });
+      
+      const response = await this.mUpdate(
+        index,
+        collection,
+        documents,
+        { refresh }
+      );
+
+      debug('Update by query: %o', esRequest);
+
+      await this.refreshCollection(index, collection);
+
+      return {
+        documents: response.items,
+        total: response.items.length,
```

Comment:
I don't see the point to return `total` and `updated` if it's the same value

Reply:
Well, i'm not sure about that, see the ES doc
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html